### DISCUSSION TITLE
How do I start a UI?

### DISCUSSION THREAD

**Discussion by corpix** [filecoin-project/boost#1683] [General]
Tutorial is great. But how do I start the UI? `boostd run` successfully started, but it does not listen any port, did I missed something?

**Comment by LexLuthr** [comment]
@corpix I would need some details.
1. Boost version
2. Did you migrate from lotus-market or is this a new Boost node
3. If migrated then which documentation was followed for migration or installation?
4. I would need your build command used to compile the binaries.
5. Please also attach the boostd start up logs

**Reply by corpix** [reply]
1. boost version 2.0.0
2. no, I am setting up a local-testnet https://docs.filecoin.io/networks/local-testnet with lotus version 1.23.3+2k, also I have used this docs to build boost https://boost.filecoin.io/getting-started
3. -
4. thats could be a bit complicated because I am packaging lotus & boost for Nix package manager, here is how I build boost https://gist.github.com/corpix/cb6469a618c3222c4ab8fc13ba9e469b#file-lotus-boost-nix-L145-L155
5. I am not sure they are useful because logs spammed with:
```
Sep 07 23:08:45 apollo filecoin-boostd-start[117647]: [64B blob data]
Sep 07 23:08:45 apollo filecoin-boostd-start[117647]: State: idle; Current Epoch: 0; Todo: 0
Sep 07 23:08:45 apollo filecoin-boostd-start[117647]: Validated 0 messages (0 per second)
```

I think it could be me misunderstanding some concepts after reading the tutorial:

- How UI is started?
- Is this a part of boostd and I should expect it to listen on some port? If yes then should I configure something to achieve this behavior?
- While writing this I have discovered I haven't build this https://github.com/filecoin-project/boost/tree/main/react, but boostd has been build successfully, will check it

**Reply by LexLuthr** [reply]
That explains your missing UI. It listens on 127.0.0.1:8080 by default. This can be updated in the config https://boost.filecoin.io/configuration/ui-settings
You would need to build the react package to serve the UI and then rebuild `boostd` with react. 
https://boost.filecoin.io/new-boost-setup is the correct tutorial to setup a new Boost node.

**Reply by corpix** [reply]
I have built a frontend, rebuilt and restarted `boostd`, but nothing changed.
Here is what I see in the logs (`boostd run` or `boostd -vv run` - output is same):
```
Sep 08 13:11:09 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:09 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:09 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:09 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:09 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:09 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:09 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:09 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:10 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: [64B blob data]
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: State: idle; Current Epoch: 0; Todo: 0
Sep 08 13:11:11 apollo filecoin-boostd-start[123587]: Validated 8 messages (0 per second)
```

No other messages. Is this expected to log messages such this so often?
Also `boostd` listens no ports:
```console
$ ss -ntulp | grep boostd
tcp   LISTEN 0      4096       127.0.0.1:8042       0.0.0.0:*    users:(("boostd-data",pid=122395,fd=6)) 
```

**Reply by LexLuthr** [reply]
Few things so note:
1. Make sure to rebuild `boostd` after building the react app to embed it correctly in `boostd`
2. Your current problem seems to be from from the fact that you are running on devnet. Boost checked the sync status by default, we should avoid that in devnet with `--nosync=true` flag

Please take a look at docker devnet that Boost team is already shipping. https://github.com/filecoin-project/boost#running-boost-devnet-in-docker-for-development

But if your aim is to learn how to configure devnet manually instead of using it then you can carry on.

**Reply by corpix** [reply]
Running with `--nosync` made a trick.
But after starting I have discovered it tries to bind to port 8042. Which I haven't expected (because tutorial recommends `boostd-data` with yugabytedb). Let it be... probably tutorial is written for production grade boost installations, not devnet's. But I want to highlight: as a newbie with Filecoin this caused a lot of confusion for me :)

Thanks for your help, @LexLuthr !
